tests/infra_test.go

```diff
@@ -975,3 +969,32 @@ func getKeysFromMetrics(metrics map[string]float64) []string {
 	sort.Strings(keys)
 	return keys
 }
+
+func checkError(err error) error {
+	var e *neturl.Error
+	if errors.As(err, &e) && e.Timeout() {
+		return nil
+	} else {
+		return err
+	}
+}
+
+func validatedHTTPResponses(errorsChan chan error, concurrency int) error {
+	var expectedErrorsCount int = 0
+	var sampledError error
+	for ix := 0; ix < concurrency; ix++ {
+		err := <-errorsChan
+		if sampledError == nil && err != nil {
+			sampledError = checkError(err)
+			if sampledError == nil {
+				expectedErrorsCount++
+			}
+		}
+	}
+
+	if sampledError == nil && expectedErrorsCount == 0 {
+		sampledError = fmt.Errorf("timeout errors were expected due to throttling")
```

Comment:
Either return here directly or change the variable name to be generic... it is not a sampleError anymore after this assignment and it is confusing.